src/App.tsx

```diff
@@ -1,61 +1,218 @@
+      {
+        userId: selectedUser.id,
+        userName: selectedUser.name,
+        userEmail: selectedUser.email,
+        userLogin: selectedUser.username,
+        todoPosts: [
+          {
+            id: postCount,
+            title: title,
+            completed: false,
+            userId: selectedUser.id,
+            createdAt: new Date(),
+          },
+        ],
+      },
+    ]);
+  };
+
+  const handleSubmit = (event: React.FormEvent) => {
+    event.preventDefault();
+
+    if (formHasSomeError) {
+      return;
+    }
+
+    setTitle('');
+    setUserSelectingValue(0);
+    setHasTitleAnError(false);
+    setHasSelectedUserAnError(false);
+    setPostCount(prev => prev + 1);
+    onSubmit(handleSelectedUser(userSelectingValue));
+  };
+  // #endregion
+
   return (
     <div className="App">
       <h1>Add todo form</h1>
 
-      <form action="/api/todos" method="POST">
+      <form action="/api/todos" method="POST" onSubmit={handleSubmit}>
         <div className="field">
-          <input type="text" data-cy="titleInput" />
-          <span className="error">Please enter a title</span>
+          <label>
+            {'Title:  '}
+            <input
+              type="text"
+              data-cy="titleInput"
+              placeholder="Enter a title"
+              value={title.trim()}
+              onChange={handleTitleChange}
+            />
+            {hasTitleAnError && (
+              <span className="error">Please enter a title</span>
+            )}
+          </label>
         </div>
 
         <div className="field">
-          <select data-cy="userSelect">
-            <option value="0" disabled>
-              Choose a user
-            </option>
-          </select>
+          <label>
+            {'User:  '}
+            <select
+              data-cy="userSelect"
+              value={userSelectingValue}
+              onChange={handleSelectingUser}
+            >
+              <option value="0" disabled={!isUserSelected}>
+                Choose a user
+              </option>
+              {usersFromServer.map(user => (
+                <option
+                  value={user.id}
+                  key={user.id}
+                  onClick={event =>
+                    setUserSelectingValue(+event.currentTarget.value)
+                  }
+                >
+                  {user.name}
+                </option>
+              ))}
+            </select>
 
-          <span className="error">Please choose a user</span>
+            {hasSelectedUserAnError && (
+              <span className="error">Please choose a user</span>
+            )}
+          </label>
         </div>
 
-        <button type="submit" data-cy="submitButton">
+        <button
+          type="submit"
+          data-cy="submitButton"
+          onClick={() => {
+            setHasTitleAnError(titleFieldHasAnError);
+            setHasSelectedUserAnError(selectedUserFieldHasAnError);
+          }}
```

Comment:
It's a submit button, do validations in the submit handler instead of adding additional handler to onclick

Reply:
yea, reasonably